src/validate_deepstream_vs_onnx.py

```
@@ -0,0 +1,265 @@
+
+        except Exception as e:
+            print(f"[WARN] Could not extract tensor meta in probe: {e}")
+
+        return Gst.PadProbeReturn.OK
+
+    # Attach probe on nvinfer src to capture tensor meta
+    nvinfer_src_pad = nvinfer.get_static_pad("src")
+    if nvinfer_src_pad is None:
+        print("[ERROR] Could not get nvinfer src pad")
+        return None
+    nvinfer_src_pad.add_probe(Gst.PadProbeType.BUFFER, tensor_extraction_probe, None)
+
+    pipeline.set_state(Gst.State.PLAYING)
+    loop = GLib.MainLoop()
+    Timer(5, loop.quit).start()
+    loop.run()
+    pipeline.set_state(Gst.State.NULL)
+
+    return result.get("tensor", None)
+
+
+def main(config_file, test_dir, model_name, onnx_dir, save_ds_preds=None, fail_on_missing_onnx=False):
+    test_images = sorted(
+        list(Path(test_dir).glob("*.jpg"))
+        + list(Path(test_dir).glob("*.jpeg"))
+        + list(Path(test_dir).glob("*.png"))
+    )
+    if not test_images:
+        print(f"[WARN] No images found in {test_dir}")
+
+    onnx_dir = Path(onnx_dir)
+    save_dir = Path(save_ds_preds) if save_ds_preds else None
+
+    for img in test_images:
+        print(f"[INFO] Running DeepStream on {img.name}")
+        ds_tensor = run_deepstream_inference(config_file, img)
+        if ds_tensor is None:
+            print(f"[ERROR] No tensor returned from DeepStream for {img.name} — skipping comparison")
+            continue
+
+        onnx_pred, onnx_path = load_onnx_prediction(onnx_dir, model_name, img.stem)
+        if onnx_pred is None:
+            print(f"[WARN] Missing ONNX prediction for {onnx_path}")
+            if save_dir:
+                out_file = save_ds_prediction(save_dir, model_name, img.stem, ds_tensor)
+                print(f"[INFO] Saved DeepStream prediction to {out_file} (for later alignment)")
+            if fail_on_missing_onnx:
+                print("[ERROR] Failing due to missing ONNX prediction as requested by flag.")
```

please throw an error